solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -93,9 +94,23 @@ contract RandomBeacon {
         completeGroupCreation();
     }
 
-    // function requestRelayEntry() external {
-    //     if RELAY_ENTRY_COUNT >= GROUP_FREQUENCY {
-    //         createGroup();
-    //     }
-    // }
+    function selectGroup() internal view returns (uint256) {
+        // TODO: Assert at least one group exists and implement selection logic.
+        return 1;
+    }
+
+    function requestRelayEntry(bytes calldata previousEntry) external {
+        // TODO: Require fee paid to the maintenance pool.
+        // TODO: Is nonReentrant needed?
+        relay.requestEntry(selectGroup(), previousEntry);
+    }
+
+    function submitRelayEntry(bytes calldata entry) external {
```

Comment:
I am not against using the library to better organize the code but we must ensure this function is as cheap as only possible given that we do not reimburse for submitting relay entry and there is no special reward for that. Let's keep it in mind, especially when we'll start working on more interactions (DKG, timeouts, slashing).